### PR TITLE
Implemented disconnect_player

### LINKED ISSUE #10490: Lua API: Add minetest.raw_disconnect
## Problem

I need to use `minetest.kick_player` to disconnect a player now, but it will add a `kicked` before the kick message.

## Solutions

Add `minetest.raw_discoonect(name,[reason])`, use the same format like `minetest.kick_player`

### PR DESCRIPTION
**Pull request by IceDragon200**
By reusing `kick_player`'s original implementation but removing the default prefix (i.e. Kicked) and then reimplementing `kick_player/2` in lua using `disconnect_player/2` as the backend

Fixes #10490

I do have one question, when is:

```cpp
// 13/03/15: remove this function when protocol version 25 will become
// the minimum version for MT users, maybe in 1 year
void Server::DenyAccess_Legacy(session_t peer_id, const std::wstring &reason)
```

Going to be removed, seeing as that note is over 5 years old

## To do

* [x] Test, ~~I don't have a server at the moment to test this with, unless I can disconnect myself~~ Lejo1 has confirmed it works

## How to test

```lua
-- minetest.disconnect_player(player_name, [reason])

minetest.disconnect_player("HackerMan", "we don't appreciate hackers")

-- console displays:
--   Access Denied: we don't appreciate hackers
```

### CODE REVIEW

**Comment by SmallJoker** [comment]
I don't think `DenyAccess_Legacy` should ever be removed. The alternative is an enum-Indexed "reason string" lookup table, which is awful in terms of forwards/backwards compatibility.
Just my two bits.

**Comment by IceDragon200** [comment]
In the end, it's not my call, I just saw the note while looking into this request and wanted some clarification on it.

But there is one that supports a reason code already?

```cpp
void Server::DenyAccess(session_t peer_id, AccessDeniedCode reason,
		const std::string &custom_reason)
{
	SendAccessDenied(peer_id, reason, custom_reason);
	m_clients.event(peer_id, CSE_SetDenied);
	DisconnectPeer(peer_id);
}
```

Though an improvement over that would be to have the `custom_reason` always appended and the 'reason' be renamed to `access_code` or something

**Review by Lejo1 — APPROVED** [review]
Works!

**Comment by sorcerykid** [comment]
Thanks for this PR! I think it would be nice if there was a way to forcibly disconnect the player without any message at all, essentially emulating the "Exit to Menu" button by exiting directly to the main menu.

I realize there's a small potential for abuse with such an feature, so perhaps add a minetest.conf setting such as 'disable_message_on_forced_disconnect'. This way games that provide a custom client download would be able to readily bypass the "Disconnect." message, without needing to meddle with the CPP code.

**Review by appgurueu — APPROVED** [review]
Perfect implementation

**Review by KaylebJay — APPROVED** [review]

File: doc/lua_api.txt [hunk 1]
```diff
@@ -5433,6 +5433,10 @@ Bans
 * `minetest.kick_player(name, [reason])`: disconnect a player with an optional
   reason.
     * Returns boolean indicating success (false if player nonexistant)
+* `minetest.disconnect_player(name, [reason])`: disconnect a player with an
+  optional reason, this will not prefix with 'Kicked: ' like kick_player.
+  If no reason is given defaults to 'Disconnected.'
```
Comment (rubenwardy) [inline doc/lua_api.txt:5438]:
```suggestion
  If no reason is given, it will default to 'Disconnected.'
```

**Comment by rubenwardy** [comment]
"Kicked" and "Disconnected" aren't translatable, but this isn't a new issue

**Review by rubenwardy — APPROVED** [review]

**Review by SmallJoker — APPROVED** [review]
LGTM although I don't really see a need for this change. Judging by the reactions this seems to be a desired feature, hence :+1:

**Pull request merged**